mhz16/src/main/java/com/example/upm/androidthings/driversamples/MHZActivity.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.upm.androidthings.driversamples;
+
+import android.app.Activity;
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.TextView;
+
+import com.example.upm.androidthings.driversupport.BoardDefaults;
+
+import mraa.mraa;
+
+public class MHZActivity extends Activity {
+    private static final String TAG = "MHZActivity";
+
+    upm_mhz16.MHZ16 mhz16;
+    TextView tv, tv1;
+    Runnable co2Task = new Runnable() {
+
+        @Override
+        public void run() {
+            // Moves the current thread into the background
+            android.os.Process.setThreadPriority(android.os.Process.THREAD_PRIORITY_BACKGROUND);
+            int i = 1; // iteration counter to defeat the chatty detector in Log.i
+
+            if (mhz16.setupTty())
+            {
+                try {
+                    while (true) {
+                        if (!mhz16.getData()) {
```

Comment:
There are few UART drivers which do open(), read() etc., calls for uart from UPM rather than calling MRAA api's.
This gas sensor is doing the same because of which i see open of /dev/tty1 failed: Permission denied. Out of 13 UAR sensor drivers, there are 7 drivers in UPM which does the same. Will have to talk to UPM team about this and fix it accordingly.
Meanwhile, i will try to get another sensor and write an example with it